guides/source/association_basics.md

```diff
@@ -355,2288 +447,2474 @@ class CreateAccountHistories < ActiveRecord::Migration[7.2]
+In our example, `imageable_id` could be the ID of either an `Employee` or a
+`Product`, and `imageable_type` is the name of the associated model's class, so
+either `Employee` or `Product`.
 
-If either of these saves fails due to validation errors, then the assignment statement returns `false` and the assignment itself is cancelled.
+While creating the polymorphic association manually is acceptable, it is instead
+recommended to use `t.references` or its alias `t.belong_to` and specify
+`polymorphic: true` so that Rails knows that the association is polymorphic, and
+it automatically adds both the foreign key and type columns to the table.
 
-If the parent object (the one declaring the `has_one` association) is unsaved (that is, `new_record?` returns `true`) then the child objects are not saved. They will automatically when the parent object is saved.
-
-If you want to assign an object to a `has_one` association without saving the object, use the `build_association` method.
-
-### `has_many` Association Reference
-
-The `has_many` association creates a one-to-many relationship with another model. In database terms, this association says that the other class will have a foreign key that refers to instances of this class.
+```ruby
+class CreatePictures < ActiveRecord::Migration[7.2]
+  def change
+    create_table :pictures do |t|
+      t.string :name
+      t.belongs_to :imageable, polymorphic: true
+      t.timestamps
+    end
+  end
+end
+```
 
-#### Methods Added by `has_many`
+Since polymorphic associations rely on storing class names in the database, that
+data must remain synchronized with the class name used by the Ruby code. When
+renaming a class, make sure to update the data in the polymorphic type column.
 
-When you declare a `has_many` association, the declaring class automatically gains 17 methods related to the association:
+For example, if you change the class name from `Product` to `Item` then you'd
+need to run a migration script to update the `imageable_type` column in the
+`pictures` table (or whichever table is affected) with the new class name.
+Additionally, you'll need to update any other references to the class name
+throughout your application code to reflect the change.
 
-* `collection`
-* [`collection<<(object, ...)`][`collection<<`]
-* [`collection.delete(object, ...)`][`collection.delete`]
-* [`collection.destroy(object, ...)`][`collection.destroy`]
-* `collection=(objects)`
-* `collection_singular_ids`
-* `collection_singular_ids=(ids)`
-* [`collection.clear`][]
-* [`collection.empty?`][]
-* [`collection.size`][]
-* [`collection.find(...)`][`collection.find`]
-* [`collection.where(...)`][`collection.where`]
-* [`collection.exists?(...)`][`collection.exists?`]
-* [`collection.build(attributes = {})`][`collection.build`]
-* [`collection.create(attributes = {})`][`collection.create`]
-* [`collection.create!(attributes = {})`][`collection.create!`]
-* [`collection.reload`][]
+### Models with Composite Primary Keys
```

Comment:
Done, since I spent time on the "Models with Composite Primary Keys", I updated the section in [active_record_composite_primary_keys#associations-between-models-with-composite-primary-keys](https://249b34bc.rails-docs-preview.pages.dev/guides/active_record_composite_primary_keys#associations-between-models-with-composite-primary-keys) with the changes and I referenced the section from the associations guide.